server/routes/offenceCodeDecisions/index.ts

```diff
@@ -0,0 +1,18 @@
+import express, { RequestHandler, Router } from 'express'
+import asyncMiddleware from '../../middleware/asyncMiddleware'
+import OffenceCodeDecisionsRoutes from './offenceCodeDecisions'
+import PlaceOnReportService from '../../services/placeOnReportService'
+
+export default function offenceCodeDecisionsRoutes({
+  placeOnReportService,
+}: {
+  placeOnReportService: PlaceOnReportService
+}): Router {
+  const router = express.Router()
+  const offenceCodeDecisions = new OffenceCodeDecisionsRoutes(placeOnReportService)
+  const get = (path: string, handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
+  const post = (path: string, handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
+  get('/:adjudicationNumber/:incidentRole/*', offenceCodeDecisions.view)
```

Comment:
The star matches either the id of the decision or the url of the decision (the same mechanism us used for both)

Reply:
i think it would be better to get all decisions and construct routes for each in a loop

Reply:
So instead of using the star matching?

Reply:
So I've done this and I think it looks quite a bit better! Thank you!